src/lib/ng-bootstrap-form-validation.module.ts

```diff
@@ -13,26 +13,17 @@ import { CUSTOM_ERROR_MESSAGES } from "./Tokens/tokens";
   exports: [FormValidationDirective, FormGroupComponent]
 })
 export class NgBootstrapFormValidationModule {
-  static forRoot(customErrorMessages?: ErrorMessage[]): ModuleWithProviders {
```

Comment:
It looks like the `.forRoot()` method signature no longer accepts `customErrorMessages?: ErrorMessage[]` as a param?

I think ideally we should update the `.forRoot()` method signature to take a config object so we can provide custom error message formatting configuration at the top, along with other options (such as class to use on the help blocks) and anything else we decide to add later so we don't introduce breaking changes by adding new config options. For backwards compatibility we could check to see if it's an array and assume it's the old `ErrorMessages[]`.

Reply:
What do you think of the approach that Ng-Bootstrap takes with the datepicker config: https://ng-bootstrap.github.io/#/components/datepicker/api#NgbDatepickerConfig

Reply:
https://github.com/ng-bootstrap/ng-bootstrap/blob/master/src/datepicker/datepicker-config.ts

Reply:
Seems neat. Would we still accept a top level configuration in the `.forRoot()` method?

Reply:
My thought was that we could maintain the current implementation for compatibility reasons but log a deprecation message, and then remove the functionality in a major release. While I'm not strictly against passing config via the `.forRoot` method, DI seems like a much cleaner route given Angular's ecosystem.

Reply:
I see. So it could still be configured at the top, but the configuration would got into the `providers`?

Reply:
@third774 I've reconfigured the `forRoot` method in my latest commit. We could also put a line at the top of the method like this:
```
if (customErrorMessages.length > 0) { console.error('Deprecation warning...'); }
```
I'm open to suggestions.